deploy/06_asset/06_upgrade_to_asset_v2.ts

```diff
@@ -0,0 +1,80 @@
+import {HardhatRuntimeEnvironment} from 'hardhat/types';
+import {DeployFunction} from 'hardhat-deploy/types';
+
+const func: DeployFunction = async function (
+  hre: HardhatRuntimeEnvironment
+): Promise<void> {
+  const {deployments, getNamedAccounts} = hre;
+  const {deployer, assetBouncerAdmin, assetAdmin} = await getNamedAccounts();
+  const {deploy} = deployments;
+
+  const forwarder = await deployments.get('TestMetaTxForwarder');
+
+  /**
+  const {deployments, getNamedAccounts, getChainId} = hre;
+  const chainId = await getChainId();
+
+  class Minter {
+    minter: string;
+    allowedToMint: boolean;
+    constructor(address: string, allowed: boolean) {
+      this.minter = address;
+      this.allowedToMint = allowed;
+    }
+  }
+
+  for hardhat-network testing we set some simple EOA minters as expected by tests:
+  const dummyMinter = otherAccounts[0];
+  const dummyMinter1 = otherAccounts[1];
+
+  async function getNetworkMinters(chain: string): Promise<Minter[]> {
+    const minterArray: Minter[] = [];
+    let allowed: boolean[];
+    let addresses: string[];
+
+    if (chain == '31337') {
+      addresses = [assetMinter.address, dummyMinter, dummyMinter1];
+      allowed = [true, true, true];
+      for (let i = 0; i < addresses.length; i++) {
+        minterArray.push(new Minter(addresses[i], allowed[i]));
+      }
+    } else {
+      addresses = [assetMinter.address];
+      allowed = [true];
+      for (let i = 0; i < addresses.length; i++) {
+        minterArray.push(new Minter(addresses[i], allowed[i]));
+      }
+    }
+    return minterArray;
+  }
+
+  const networkMinters = await getNetworkMinters(chainId);
+
+
+  */
+  await deploy('Asset', {
```

Comment:
For upgrade,  if we have 2 deploy function, both will be executed each time, 
so first time, the first script will see that version 1 is identical and do nothing
then the second script will upgrade to v2

but next time, the first script will attempt to revert back to v1 and second script will bring it back to v2 if not already v2


There is an optional field in the proxy option field  called  `upgradeIndex` that ensure the upgrade go through if the current deployed version used a lower upgradeIndex.